src/js/components/search/visualizations/time/TimeVisualizationSection.jsx

```diff
@@ -49,6 +59,64 @@ export default class TimeVisualizationSection extends React.Component {
         }
     }
 
+    downloadLabel = () => {
+        let periodLabel;
+        if (this.props.data.visualizationPeriod === 'fiscal_year') {
+            periodLabel = 'Year';
+        }
+        else {
+            periodLabel = capitalize(this.props.data.visualizationPeriod);
+        }
+        return `Download data by ${periodLabel}`;
+    };
+
+    downloadData = () => {
+        const data = this.props.data;
+        const ret = [];
+        if (data.visualizationPeriod === 'fiscal_year') {
+            ret[0] = ['fiscal_year', 'total_obligations'];
+        }
+        else if (data.visualizationPeriod === 'quarter') {
+            ret[0] = ['fiscal_year', 'fiscal_quarter', 'total_obligations'];
+        }
+        else {
+            ret[0] = ['fiscal_year', 'month', 'total_obligations'];
+        }
+        for (let i = 0; i < data.rawLabels.length; i++) {
+            if (data.visualizationPeriod === 'fiscal_year') {
+                ret[i + 1] = [data.rawLabels[i].year, data.ySeries[i][0]];
+            }
+            else if (data.visualizationPeriod === 'quarter') {
+                ret[i + 1] = [data.rawLabels[i].year, data.rawLabels[i].period[1], data.ySeries[i][0]];
+            }
+            else {
+                const month = data.rawLabels[i].period;
+                ret[i + 1] = [data.rawLabels[i].year, fullMonthFromAbbr(month), data.ySeries[i][0]];
+                if (['Oct', 'Nov', 'Dec'].indexOf(month) > -1) {
+                    // correct FY
+                    ret[i + 1][0] = parseInt(ret[i + 1][0], 10) + 1;
+                }
+            }
+        }
+        return ret;
+    };
+
+    downloadCsv = () => {
+        let contents = this.downloadData();
```

Comment:
we shouldn't use `let` here. Mutable data is troublesome.

Reply:
true